src/core/graph.js

```diff
@@ -889,6 +889,71 @@ export class Graph {
     return result;
   }
 
+  /**
+   * Create a new graph, in which some nodes have been combined.
+   *
+   * combineNodes allows collapsing nodes together. For example, if we had a graph
+   * with nodes A, B and C, then we could collapse A into B, creating a new
+   * graph with just nodes B and C. The edges in the graph are re-written in concert;
+   * for example, an edge from A to B would become a loop edge from B to itself.
+   *
+   * combineNodes' motivating use cases is collapsing different user identities together.
+   * For example, if a user is represented on both GitHub and Discourse, we can create
+   * a new node that represents that identity, and collapse their GitHub and Discourse user
+   * nodes into it. (If we chose, we could also collapse the GitHub node into the Discourse
```

Comment:
The form that always contracts into a new node rather than choosing one
of the nodes arbitrarily is certainly nicer in some ways. For example,
suppose that the GitHub plugin expects its authorship edges to always be
connected to users. If we contract arbitrarily, then there’s a “50/50
chance” that the GitHub plugin will break, and so an error might slink
by; if we always create a new node, we’ll be sure to discover the
problem immediately.

Was there any consideration for the signature

```js
type Contraction = {|
  +old: $ReadOnlyArray<NodeAddressT>,
  +new: NodeAddressT,
|};

contractNodes(contractions: $ReadOnlyArray<Contraction>): Graph {
  // …
}
```

where `new` fresh?